using/src/activations.rs

```diff
@@ -0,0 +1,178 @@
+//! Methods to select which of a set of implementations to activate based on a schema's
+//! `Request`s.
+
+use crate::{Find, Found, Implementations, Request, Schema};
+
+/// An Activation references a request and contains the min and max versioned implementations
+/// available in the implementation registry which can satisfy that request, or None if no
+/// matching implementations were available.
+// type Activation<'schema, 'impls, T> = (&'schema Request, Find<'impls, T>);
+
+impl<'a> Schema<'a> {
+    /// Given a set of implementations, take all requests in the document and
+    /// match them to the implementations which satisfy them ("satisfying
+    /// implementations").
+    ///
+    /// This returns an `Iterator` over `(&Request, Find<T>)`.
+    /// `Find` is an `Iterator` over `(&Version, &T)`, which will iterate satisfying
+    /// implementations ordered by lowest to highest version. The
+    /// [`bound`](Bounds.html#bound) method can be used to get the lowest and highest
+    /// satisfying versions.
+    ///
+    /// # Example:
+    ///
+    /// ```
+    /// use using::*;
+    ///
+    /// let impls = Implementations::new()
+    ///     .provide("https://spec.example.com/A", Version(1, 2), "impl for A 1.2")
+    ///     .provide("https://spec.example.com/A", Version(1, 3), "impl for A 1.3")
+    ///     .provide("https://spec.example.com/A", Version(1, 7), "impl for A 1.7");
+    /// let schema = Schema::parse(r#"schema @using(spec: "https://spec.example.com/A/v1.0") { query: Query }"#)?;
+    ///
+    /// let max = schema
+    ///   .activations(&impls)
+    ///   .filter_map(|(_request, mut impls)| impls.bounds())
+    ///   .collect::<Vec<_>>();
+    ///
+    /// assert_eq!(max, vec![(
+    ///     (&Version(1, 2), &"impl for A 1.2"), (&Version(1, 7), &"impl for A 1.7")
+    /// )]);
+    /// # Ok::<(), GraphQLParseError>(())
+    /// ```
+    pub fn activations<'impls, T>(
+        &'impls self,
```

Comment:
With `impl` being a reserved word, it's a bit weird to have `'impls` as a name for a lifetime. Would definitely use something else (I do understand it's trying to describe a lifetime of Implementations, which is just very close to `impl` hah)